Feature/accessible archive widget

RobjS opened the pull request:
The default WordPress "Archives" widget is not accessible if you use the dropdown format, as it uses JavaScript to load the page as soon as the user makes a choice from the dropdown. This PR adds a custom "Archives dropdown with submit" widget that should provide an accessible alternative.

A few things to note:

* This is the first iteration, so doesn't offer any of the options the main widget has. In particular, you can't change the title (it's always just "Archives"), and the list of archive pages is always monthly, and always includes the post count per month in brackets
* The "Go" button this widget includes (to activate the choice made in the dropdown) is still JS-dependent. Therefore, if the user has JS deactivated, we're replacing the dropdown & button with a `<ul>` of links to the archive pages.
* The tests for both this widget and the accessible categories widget this is largely based on aren't great, as they test the exact HTML output, which means they're very brittle. They could really do with a substantial refactor, but that's beyond the scope of this bit of work.


Example with JS switched on:

<img width="310" alt="Screenshot 2021-02-11 at 11 46 28" src="https://user-images.githubusercontent.com/370665/107632722-ca330900-6c5e-11eb-939c-ee64f6490414.png">


With JS switched off:

<img width="226" alt="Screenshot 2021-02-11 at 11 46 57" src="https://user-images.githubusercontent.com/370665/107632777-db7c1580-6c5e-11eb-931d-f874af974472.png">

calumryan commented:
@RobjS this looks good 👍🏻 

I'd suggest we use the same class names from the govuk design system for the select 
`govuk-select` and screen reader-only text on label as `govuk-visually-hidden`. 

I'll probably add some additional styling after too for the alignment of the button.

calumryan commented:
Non-js alternative for list e.g. https://design-system.service.gov.uk/styles/typography/#lists

RobjS commented:
Thanks @calumryan, I've modified the classes where I can to use the standard `govuk-` ones.

I've also removed a bit of unnecessary styling on the pre-existing categories widget so it's dropdown also looks like the default `govuk-select` one.

Unfortunately it's not straightforward to modify the markup WordPress outputs for the `<ul>` of categories, so we might have to do without the `govuk-list` classes on that.
